vmslave-config/pipeline/vmslave_config.groovy

```diff
@@ -0,0 +1,29 @@
+def checkout(String url, String branch, String targetDir){
+    checkout(
+    [$class: 'GitSCM', branches: [[name: branch]],
+    extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: targetDir]],
+    userRemoteConfigs: [[url: url]]])
+}
+
+node('vmslave-config') {
```

Comment:
can you leverage pipeline power, to dynamic provision slave with same label like "pr-gate" and update the 'host' file